tst/com/osreboot/ridhvl2/TestMath.java

```diff
@@ -39,4 +39,14 @@ public void testLimit(){
 		assertEquals(10, HvlMath.limit(10, 10, 10), 0);
 	}
 	
+	@Test
+	public void testRandomInt() {
+		assertEquals(1, HvlMath.randomInt(1, 1),0);
+		int testAccumulator = 0;
+		for(int i=0; i<1000; i++) {
+			testAccumulator += HvlMath.randomInt(0, 10);
+		}
+		assertEquals(5, testAccumulator/1000, 1);
+	}
+	

```

Comment:
I _guess_ this could give you a result that indicates whether or not the boundaries are accurate, but it seems like a roundabout way of doing it and I'm not able to say for sure if what you've written will accurately test the boundaries (at least I can't verify from just reading this code, which is problematic).

Instead try this;
1. Create a HashSet of integers that represent every different output the random function has produced.
2. Run the function ~10000 times and put every result into the HashSet. Because sets don't contain duplicates this will record every _different_ output that was produced.
3. Loop over the entire possible range of output integers and assert that the HashSet includes them.

Reply:
Also I'd do this entire thing twice, once for a range like 2-5 and once for that range reversed, i.e. 5-2 so we can verify that both orders work.